### PR TITLE
Review LGPL-2.1 license headers

### PR DESCRIPTION
**Pull request by Adrian-Stanea**
## Introduction 
This PR standardizes the license headers and fixes version mismatches, ensuring all files use `LGPL v2.1` instead of `v2` or `v3`.

## To be discussed
1. These files are currently under the more restrictive ADI-BSD license. They should be reviewed to either replace them with alternatives under a more permissive license or to ensure proper mention of their license in the documentation or project files.
    - [x] ~~include/libm2k/tools/i2c.hpp~~
    - [x] ~~include/libm2k/tools/spi.hpp~~
    - [x] ~~include/libm2k/tools/uart.hpp~~
2. Consider adding a CI check to ensure that certain files contain the appropriate license headers. The CI check should fail if any files do not follow the established license guidelines.

## Note

The following files and directories have been omitted from the reviewed files:
### Directories:
  - `.git*`
  - `build`
  - `CI`
  - `img`
  - `results`
  - `wingetopt`
 
### Files:
  - `*.png`
  - `*.rst`
  - `.gitignore`
  - `53-adi-m2k-usb.rules`
  - `AssemblyInfo.cs.in`
  - `azure-pipelines.yml`
  - `Doxyfile.in`
  - `Doxyfile_API.in`
  - `README*`
  - `key.snk`
  - `libm2k-sharp.dll.config.cmakein`
  - `libm2k-sharp.pc.cmakein`
  - `libm2k.i`
  - `libm2k.pc.cmakein`
  - `libm2kData.xml`
  - `requirements.txt`
  - `version.hpp.cmakein`

### CODE REVIEW

File: bindings/labview/libm2k_lv.cpp [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
 #include "libm2k_lv.h"
 #include <libm2k/m2kexceptions.hpp>
-#include <libm2k/contextbuilder.hpp>
+#include <libm2k/contextbuilder.hppbindings/libm2k.i>
```
Comment (AlexandraTrifan) [inline bindings/labview/libm2k_lv.cpp:24]: i think this is a unwanted change

**Review by AlexandraTrifan — APPROVED** [review]
Otherwise looks good. 
Let's wait for @adisuciu to take a look at this also.
And for now we'll keep this PR open until we have a clear overview of the tools/i2c spi and uart files you mentioned earlier.

It would indeed be a good idea to add a CI job to run a script and do a minimal checkup of the license header existence and version.

**Review by AlexandraTrifan — CHANGES_REQUESTED** [review]
later edit: We should not modify the license headers in files from deps/wingetopt

File: bindings/labview/libm2k_lv.cpp (hunk 1 above)
Comment (Adrian-Stanea) [reply bindings/labview/libm2k_lv.cpp:24]: Removed

**Comment by Adrian-Stanea** [comment]
> later edit: We should not modify the license headers in files from deps/wingetopt

Removed license header from cmake file from tools/deps/wingetopt

**Comment by Adrian-Stanea** [comment]
Replaced ADI-BSD license

**Review by AlexandraTrifan — APPROVED** [review]

**Pull request merged**